java/ql/src/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -138,127 +188,420 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ";CollectionUtils;true;union;;;Element of Argument[1];Element of ReturnValue;value",
+          ";CollectionUtils;true;unmodifiableCollection;;;Element of Argument[0];Element of ReturnValue;value"
+        ]
+  }
+}
+
+/**
+ * Value-propagating models for the class `org.apache.commons.collections4.ListUtils`.
+ */
+private class ApacheListUtilsModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      ["org.apache.commons.collections4", "org.apache.commons.collections"] +
+        [
+          ";ListUtils;true;defaultIfNull;;;Argument[0];ReturnValue;value",
+          ";ListUtils;true;defaultIfNull;;;Argument[1];ReturnValue;value",
+          ";ListUtils;true;emptyIfNull;;;Argument[0];ReturnValue;value",
+          ";ListUtils;true;fixedSizeList;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;intersection;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;intersection;;;Element of Argument[1];Element of ReturnValue;value",
+          ";ListUtils;true;lazyList;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;longestCommonSubsequence;(CharSequence,CharSequence);;Argument[0];ReturnValue;taint",
+          ";ListUtils;true;longestCommonSubsequence;(CharSequence,CharSequence);;Argument[1];ReturnValue;taint",
+          ";ListUtils;true;longestCommonSubsequence;(List,List);;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;longestCommonSubsequence;(List,List);;Element of Argument[1];Element of ReturnValue;value",
+          ";ListUtils;true;longestCommonSubsequence;(List,List,Equator);;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;longestCommonSubsequence;(List,List,Equator);;Element of Argument[1];Element of ReturnValue;value",
+          ";ListUtils;true;partition;;;Element of Argument[0];Element of Element of ReturnValue;value",
+          ";ListUtils;true;predicatedList;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;removeAll;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;retainAll;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;select;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;selectRejected;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;subtract;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;sum;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;sum;;;Element of Argument[1];Element of ReturnValue;value",
+          ";ListUtils;true;synchronizedList;;;Element of Argument[0];Element of ReturnValue;value",
+          ";ListUtils;true;transformedList;;;Element of Argument[0];Element of ReturnValue;value",
```

Comment:
Probably worth a comment on this and `lazyList` that these models represent the perhaps-surprising behaviour that existing list elements are retained unaltered by both methods